src/Orleans.Core/Runtime/SingleTaskAsynchAgent.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using System.Threading;
+using Microsoft.Extensions.Logging;
+
+namespace Orleans.Runtime
+{
+    internal abstract class SingleTaskAsynchAgent : AsynchAgent
+    {
+        protected SingleTaskAsynchAgent(string nameSuffix, ExecutorService executorService, ILoggerFactory loggerFactory) : base(nameSuffix, executorService, loggerFactory)
+        {
+        }
+
+        protected SingleTaskAsynchAgent(ExecutorService executorService, ILoggerFactory loggerFactory) : base(executorService, loggerFactory)
+        {
+        }
+
+        public override void OnStart()
+        {
+            executor.QueueWorkItem(_ => AgentThreadProc(this));
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
+        private void AgentThreadProc(Object obj)
```

Comment:
This could be made static, since `obj` = `this`, otherwise no need to use `obj` at all. If using `obj`, it should be passed as the context parameter to `QueueWorkItem` rather than closed over.

None of this matters from a perf perspective since this is called so infrequently.

Reply:
Fixed